### ISSUE TITLE
[Bug] Contact block's phone number field uses an invalid phone number autocomplete value (phone)

### ISSUE THREAD

**Issue by EricDunsworth** [cds-snc/platform-forms-client#3048]
Spinoff of #3047.

The contact block's phone number field uses an ``autocomplete="phone"`` attribute. That makes sense given the field's purpose.

But no such value exists in HTML. The closest match for it in the [WHATWG HTML spec's ``autocomplete`` section](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofilling-form-controls:-the-autocomplete-attribute) is [``autocomplete="tel"``](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fe-autocomplete-tel).

**Reproduction steps:**
1. Visit the [form builder editor](https://forms-staging.cdssandbox.xyz/form-builder/edit)
2. Press the "+ Add" button, pick "Contact" and press the "Select block" button
3. Switch to the "Preview" tab
4. Open your browser's DOM inspector
5. Use the inspector to highlight the "Phone number" form field

**Expected result:**
Step 5 should show an ``input type="tel"`` element with an ``autocomplete="tel"`` attribute.

**Actual result:**
Step 5 shows an ``input type="text"`` element with an ``autocomplete="phone"`` attribute.

That ``autocomplete`` value is meaningless to user agents. [HTML validation](https://validator.w3.org/nu/) and [WCAG 2.x SC 1.3.5](https://www.w3.org/WAI/WCAG22/Understanding/identify-input-purpose.html) (via [F107](https://www.w3.org/WAI/WCAG22/Techniques/failures/F107)) also fail.

**Proposed solution:**
*  Use ``autocomplete="tel"`` instead of autocomplete="phone" for the contact block's "Phone number" field.
* If any phone number validation is in play (I can't test it in the preview tab)... ensure it treats [``autocomplete="tel"``'s canonical format](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fe-autocomplete-tel) as valid.
* Consider using using an ``input type="tel"`` element instead of ``input type="text"`` in this scenario. It'll help user agents programmatically-determine the field's purpose - which may have other benefits (like causing mobile browsers to show virtual phone number keyboards instead of "generic" ones). Not to mention that it'd be more in line with what the email field is already doing (it uses its namesake ``type`` - ``input type="email"``).

**Comment by thiessenp-cds** [comment]
Fixed in related PR #3077
